Turn on phone number of HQ signup AB test

mkaczmarczyksoldevelo opened the pull request.

mkaczmarczyksoldevelo committed:
Turn on phone number of HQ signup AB test

orangejenny commented:
The data currently being sent with events should be attached to the user, not to the events themselves. You can use a call to `identifyTraits` for this.

mkaczmarczyksoldevelo commented on corehq/apps/registration/static/registration/js/new_user.ko.js:
```diff
@@ -298,6 +301,12 @@ hqDefine('registration/js/new_user.ko', function () {
                     self.currentStep(_nextStep);
                 });
 
+            if (_nextStep === 1) {
```
@orangejenny did that

mkaczmarczyksoldevelo commented:
@orangejenny for `identifyTraits` I have added this to the second event because the first is sent when the user visits the register website before he/she have provided an email. In the second step, I've added the `identifyTraits`.

orangejenny commented:
Hey, the updates look good, but I think calling `identifyTraits` is sufficient, I don't think the data also needs to be passed along to the events.

mkaczmarczyksoldevelo commented:
@orangejenny fixed accordingly to our skype talk.

orangejenny commented on corehq/apps/registration/static/registration/js/new_user.ko.js:
```diff
@@ -38,11 +38,14 @@ hqDefine('registration/js/new_user.ko', function () {
     _kissmetrics.whenReadyAlways(function() {
-        _private.isAbPhoneNumber = _kissmetrics.getAbTest('New User Phone Number') === 'show_number';
+        _private.isAbPhoneNumberShown = _kissmetrics.getAbTest('New User Phone Number') === 'show_number';
+        _kissmetrics.track.event("Viewed CommCare signup page", {
```
This event doesn't need the data, either.

mkaczmarczyksoldevelo replied:
ok, so `identifyTraits` it is
